### ISSUE TITLE
Paginators return error when next token is not present

### ISSUE THREAD

**Issue by imp** [awslabs/aws-sdk-rust#405]
### What is the problem?

While migrating my code that does various `ec2::describe_XXX` calls  to use the new paginator available in `0.4.x` I am hitting the
```
Error: failed to construct request: next token did not change, aborting paginator. This indicates an SDK or AWS service bug.
```

It feels like it happens when there are only a few items to fetch from the service, but I have no clear proof for it. Looping manually passing tokens from the previous call to the next one works as expected.

### Version

```
├── aws-config v0.4.1
│   ├── aws-http v0.4.1
│   │   ├── aws-smithy-http v0.34.1
│   │   │   ├── aws-smithy-types v0.34.1
│   │   ├── aws-smithy-types v0.34.1 (*)
│   │   ├── aws-types v0.4.1
│   │   │   ├── aws-smithy-async v0.34.1
│   │   │   ├── aws-smithy-types v0.34.1 (*)
│   ├── aws-sdk-sts v0.4.1
│   │   ├── aws-endpoint v0.4.1
│   │   │   ├── aws-smithy-http v0.34.1 (*)
│   │   │   ├── aws-types v0.4.1 (*)
│   │   ├── aws-http v0.4.1 (*)
│   │   ├── aws-sig-auth v0.4.1
│   │   │   ├── aws-sigv4 v0.4.1
│   │   │   │   ├── aws-smithy-http v0.34.1 (*)
│   │   │   ├── aws-smithy-http v0.34.1 (*)
│   │   │   ├── aws-types v0.4.1 (*)
│   │   ├── aws-smithy-async v0.34.1 (*)
│   │   ├── aws-smithy-client v0.34.1
│   │   │   ├── aws-smithy-async v0.34.1 (*)
│   │   │   ├── aws-smithy-http v0.34.1 (*)
│   │   │   ├── aws-smithy-http-tower v0.34.1
│   │   │   │   ├── aws-smithy-http v0.34.1 (*)
│   │   │   ├── aws-smithy-types v0.34.1 (*)
│   │   ├── aws-smithy-http v0.34.1 (*)
│   │   ├── aws-smithy-http-tower v0.34.1 (*)
│   │   ├── aws-smithy-query v0.34.1
│   │   │   ├── aws-smithy-types v0.34.1 (*)
│   │   ├── aws-smithy-types v0.34.1 (*)
│   │   ├── aws-smithy-xml v0.34.1
│   │   ├── aws-types v0.4.1 (*)
│   ├── aws-smithy-async v0.34.1 (*)
│   ├── aws-smithy-client v0.34.1 (*)
│   ├── aws-smithy-http v0.34.1 (*)
│   ├── aws-smithy-http-tower v0.34.1 (*)
│   ├── aws-smithy-json v0.34.1
│   │   └── aws-smithy-types v0.34.1 (*)
│   ├── aws-smithy-types v0.34.1 (*)
│   ├── aws-types v0.4.1 (*)
├── aws-sdk-cloudformation v0.4.1
│   ├── aws-endpoint v0.4.1 (*)
│   ├── aws-http v0.4.1 (*)
│   ├── aws-sig-auth v0.4.1 (*)
│   ├── aws-smithy-async v0.34.1 (*)
│   ├── aws-smithy-client v0.34.1 (*)
│   ├── aws-smithy-http v0.34.1 (*)
│   ├── aws-smithy-http-tower v0.34.1 (*)
│   ├── aws-smithy-query v0.34.1 (*)
│   ├── aws-smithy-types v0.34.1 (*)
│   ├── aws-smithy-xml v0.34.1 (*)
│   ├── aws-types v0.4.1 (*)
├── aws-sdk-ec2 v0.4.1
│   ├── aws-endpoint v0.4.1 (*)
│   ├── aws-http v0.4.1 (*)
│   ├── aws-sig-auth v0.4.1 (*)
│   ├── aws-smithy-async v0.34.1 (*)
│   ├── aws-smithy-client v0.34.1 (*)
│   ├── aws-smithy-http v0.34.1 (*)
│   ├── aws-smithy-http-tower v0.34.1 (*)
│   ├── aws-smithy-query v0.34.1 (*)
│   ├── aws-smithy-types v0.34.1 (*)
│   ├── aws-smithy-xml v0.34.1 (*)
│   ├── aws-types v0.4.1 (*)
```

### Platform

Darwin hellride 21.2.0 Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64 x86_64

### AWS Services

I have tried this with ec2 and cloudformation

### Description

Here is the code snippet that produces the error:
```
        self.vpcs = self
            .client
            .describe_vpcs()
            .into_paginator()
            .items()
            .send()
            .collect::<Result<_, _>>()
            .await?;
```

### Logs

<details>
<summary>Response from the AWS service</summary>
```
2022-01-18T15:01:56.863335Z TRACE aws_smithy_http::middleware: http_response=Response { status: 200, version: HTTP/1.1, headers: {"x-amzn-requestid": "ef502f55-3764-4362-a0d0-be5090febfce", "cache-control": "no-cache, no-store", "strict-transport-security": "max-age=31536000; includeSubDomains", "vary": "accept-encoding", "content-type": "text/xml;charset=UTF-8", "content-length": "2427", "date": "Tue, 18 Jan 2022 15:01:56 GMT", "server": "AmazonEC2"}, body: b"<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<DescribeVpcsResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-11-15/\">\n    <requestId>ef502f55-374-4332-a0c0-bc6090febfce</requestId>\n    <vpcSet>\n        <item>\n            <vpcId>vpc-xxxxxxxxxxxxxxxxx</vpcId>\n            <ownerId>xxxxxxxxxxxx</ownerId>\n            <state>available</state>\n            <cidrBlock>10.16.1.192/28</cidrBlock>\n            <cidrBlockAssociationSet>\n                <item>\n                    <cidrBlock>10.16.1.192/28</cidrBlock>\n                    <associationId>vpc-cidr-assoc-xxxxxxxxxxxxxxxxx</associationId>\n                    <cidrBlockState>\n                        <state>associated</state>\n                    </cidrBlockState>\n                </item>\n                <item>\n                    <cidrBlock>10.112.1.192/28</cidrBlock>\n                    <associationId>vpc-cidr-assoc-xxxxxxxxxxxxxxxxx</associationId>\n                    <cidrBlockState>\n                        <state>associated</state>\n                    </cidrBlockState>\n                </item>\n                <item>\n                    <cidrBlock>10.16.1.208/28</cidrBlock>\n                    <associationId>vpc-cidr-assoc-xxxxxxxxxxxxxxxxx</associationId>\n                    <cidrBlockState>\n                        <state>associated</state>\n                    </cidrBlockState>\n                </item>\n                <item>\n                    <cidrBlock>10.112.1.208/28</cidrBlock>\n                    <associationId>vpc-cidr-assoc-xxxxxxxxxxxxxxxxx</associationId>\n                    <cidrBlockState>\n                        <state>associated</state>\n                    </cidrBlockState>\n                </item>\n            </cidrBlockAssociationSet>\n            <dhcpOptionsId>dopt-xxxxxxxx</dhcpOptionsId>\n            <tagSet>\n                <item>\n                    <key>sigs.k8s.io/cluster-api-provider-aws/cluster/mgmt2</key>\n                    <value>owned</value>\n                </item>\n                <item>\n                    <key>sigs.k8s.io/cluster-api-provider-aws/role</key>\n                    <value>common</value>\n                </item>\n                <item>\n                    <key>Name</key>\n                    <value>mgmt2-vpc</value>\n                </item>\n            </tagSet>\n            <instanceTenancy>default</instanceTenancy>\n            <isDefault>false</isDefault>\n        </item>\n    </vpcSet>\n</DescribeVpcsResponse>" }
```
</details>

It seems there is no next token at all (implying there is no next page and all the requested data is present?)

**Comment by rcoh** [comment]
thanks for the report! I see the problem and will get a fix out soon

**Comment by rcoh** [comment]
The fix has landed, you can test it out prior to the the next release with:
```
# you'll also need the matching version of aws-config
aws-config = { git = "https://github.com/awslabs/aws-sdk-rust", branch = "next" }

aws-sdk-ec2 = { git = "https://github.com/awslabs/aws-sdk-rust", branch = "next" }
```

**Comment by imp** [comment]
I can confirm it works with `next` branch. Thanks for addressing the issue quickly.
If it is possible to publish a fixed drop to crates.io it would be fantastic!

**Comment by rcoh** [comment]
Keeping the issue open until it's in a release in case other folks run into it.

A new release should go out soon

**Comment by rcoh** [comment]
landed in 0.5.2
